### PR TITLE
Remove beta indicators from work pool docs

### PR DESCRIPTION
**Pull request by billpalombi**
- Removes beta indicators from work pool docs
- Moves content related to agents to the bottom of the page
- Minor edits for consistency

### CODE REVIEW

File: docs/concepts/work-pools.md [hunk 1]
```diff
@@ -14,112 +14,24 @@ search:
   boost: 2
 ---
 
-# Work Pools, Workers & Agents
+# Work Pools, Workers, & Agents
```
Comment (jawnsy) [inline docs/concepts/work-pools.md:17]: nitpick: I'd propose we spell out "and"

Comment (billpalombi) [reply docs/concepts/work-pools.md:17]: agreed that's better

**Comment by zhen0** [comment]
@billpalombi - do we want to add this for today's release?

**Comment by billpalombi** [comment]
Yes!

**Review by zhen0 — APPROVED** [review]
Thanks @billpalombi

**Pull request merged**